drivers/hwinfo/hwinfo_andes.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2022 Andes Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/hwinfo.h>
+#include <zephyr/drivers/syscon.h>
+#include <string.h>
+
+/*
+ * SMU(System Management Unit) Registers for hwinfo driver
+ */
+
+/* Register offset*/
+#define SMU_SYSTEMVER	0x00
+#define SMU_WRSR	0x10
+
+/* Wake-Up and Reset Status Register bitmask */
+#define SMU_WRSR_APOR	BIT(0)
+#define SMU_WRSR_MPOR	BIT(1)
+#define SMU_WRSR_HW	BIT(2)
+#define SMU_WRSR_WDT	BIT(3)
+#define SMU_WRSR_SW	BIT(4)
+
+#define ANDES_RESET_STATUS_MASK	BIT_MASK(5)
+
+ssize_t z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length)
+{
+	int ret = 0;
+	uint8_t id[3];
+	uint32_t ver;
+	const struct device *const syscon_dev =
+			DEVICE_DT_GET(DT_NODELABEL(syscon));
```

Comment:
suggestion: make it a global (static) variable. You should also check for readiness (maybe once in a SYS_INIT, using asserts here...)